titiler/api/endpoints/mosaic.py

```diff
@@ -126,3 +145,77 @@ def mosaicjson_info(
             response["quadkeys"] = list(mosaic_quadkeys)
             response = {**asset_info, **response}
         return response
+
+
+@router.get(r"/tiles/{z}/{x}/{y}", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}\.{format}", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}@{scale}x", **tile_response_codes)
+@router.get(r"/tiles/{z}/{x}/{y}@{scale}x\.{format}", **tile_response_codes)
+async def mosaic_tile(
+    z: int = Path(..., ge=0, le=30, description="Mercator tiles's zoom level"),
+    x: int = Path(..., description="Mercator tiles's column"),
+    y: int = Path(..., description="Mercator tiles's row"),
+    scale: int = Query(
+        1, gt=0, lt=4, description="Tile size scale. 1=256x256, 2=512x512..."
+    ),
+    format: ImageType = Query(None, description="Output image type. Default is auto."),
+    url: str = Query(..., description="Cloud Optimized GeoTIFF URL."),
+):
+    """Read MosaicJSON tile"""
+    # TODO: Maybe use ``read_mosaic`` defined above (depending on cache behavior which is still TBD)
+    with MosaicBackend(url) as mosaic:
+        assets = mosaic.tile(x=x, y=y, z=z)
+
+    tilesize = 256 * scale
+
+    # TODO: Parametrize pixsel method
+    pixel_selection = FirstMethod()
+
+    # Rio-tiler-mosaic uses an external ThreadPoolExecutor to process multiple assets at once but we want to use the
+    # executor provided by the event loop.  Instead of calling ``rio_tiler_mosaic.mosaic.mosaic_tiler`` directly we will
+    # transcribe the code here and use the executor provided by the event loop.  This also means we define this function
+    # as a coroutine (even though nothing that is called is a coroutine), since the event loop's executor isn't
+    # available in normal ``def`` functions.
+    # https://github.com/cogeotiff/rio-tiler-mosaic/blob/master/rio_tiler_mosaic/mosaic.py#L37-L102
+    _tiler = partial(cogeoTiler, tile_x=x, tile_y=y, tile_z=z, tilesize=tilesize)
+    futures = [run_in_threadpool(_tiler, asset) for asset in assets]
+
+    # TODO: parametrize concurrency
+    semaphore = asyncio.Semaphore(10)
```

Comment:
Semaphore is used to limit concurrency instead of [chunking list of assets](https://github.com/cogeotiff/rio-tiler-mosaic/blob/master/rio_tiler_mosaic/mosaic.py#L90-L91)